scripts/list_outdated_dependencies.py

```diff
@@ -0,0 +1,272 @@
+
+
+def remove_self_dependencies(pyproject: dict) -> dict:
+    """Filter out self dependencies (dependencies of the package on it self) from the
+    dependencies and optional-dependencies in the provided pyproject metadata."""
+
+    if "project" not in pyproject:
+        return pyproject
+
+    modified_pyproject = deepcopy(pyproject)
+
+    project_metadata = modified_pyproject["project"]
+
+    package_name = stringcase.spinalcase(project_metadata.get("name"))
+
+    if not package_name:
+        raise ValueError("The provided project metadata does not contain a name.")
+
+    if "dependencies" in project_metadata:
+        project_metadata["dependencies"] = exclude_from_dependency_list(
+            package_name=package_name, dependencies=project_metadata["dependencies"]
+        )
+
+    if "optional-dependencies" in project_metadata:
+        for group in project_metadata["optional-dependencies"]:
+            project_metadata["optional-dependencies"][
+                group
+            ] = exclude_from_dependency_list(
+                package_name=package_name,
+                dependencies=project_metadata["optional-dependencies"][group],
+            )
+
+    return modified_pyproject
+
+
+def get_main_deps_pyproject(modified_pyproject: dict[str, Any]) -> list[str]:
+    """Get a list of the dependencies from pyproject.toml"""
+
+    dependencies: list[str] = []
+    if "dependencies" in modified_pyproject["project"]:
+        dependencies = modified_pyproject["project"]["dependencies"]
+
+    return dependencies
+
+
+def get_optional_deps_pyproject(modified_pyproject: dict[str, Any]) -> list[str]:
+    """Get a list of the optional dependencies from pyproject.toml"""
+
+    dependencies: list[str] = []
+
+    if "optional-dependencies" in modified_pyproject["project"]:
+        for optional_dependency_list in modified_pyproject["project"][
+            "optional-dependencies"
+        ]:
+            dependencies.extend(
+                modified_pyproject["project"]["optional-dependencies"][
+                    optional_dependency_list
+                ]
+            )
+
+    return dependencies
+
+
+def get_modified_pyproject() -> dict[str, Any]:
+    """Get a copy of pyproject.toml with any self-referencing dependencies removed."""
+    with open(PYPROJECT_TOML_PATH, "rb") as pyproject_toml:
+        pyproject = tomli.load(pyproject_toml)
+
+    modified_pyproject = remove_self_dependencies(pyproject)
+    return modified_pyproject
+
+
+def get_deps_dev() -> list[str]:
+    """Get a list of raw dependency strings from requirements-dev.in"""
+    dependencies = []
+    with open(DEV_DEPS_PATH, encoding="utf-8") as dev_deps:
+        lines = dev_deps.readlines()
+
+    for line in lines:
+        if (
+            line.strip().startswith("#")  # skip comments
+            or len(line.strip()) == 0  # skip empty lines
+            or "requirements-dev-common.in" in line  # skip the inclusion line
+        ):
+            continue
```

Comment:
```suggestion
        line = line.strip()
        if (
            not line  # skip empty lines
            or line.startswith("#")  # skip comments
            or "requirements-dev-common.in" in line  # skip the inclusion line
        ):
            continue
```

Reply:
or maybe

```python
dependencies = [
    line for line in (line.strip() for line in dev_deps)
    if line  # skip empty lines 
       and not line.startswith("#")  # skip comments
       and line != "requirements-dev-common.in"  # skip inclusion line
]
```

Reply:
Nice, went with that in [3c50ba4](https://github.com/ghga-de/microservice-repository-template/pull/165/commits/3c50ba4af424dd9cf46e190572ffdddc7527bdca)